docs/core/porting/cpp-cli-alts.md

```diff
@@ -0,0 +1,175 @@
+
+Tooling, such as [ClangSharp](https://github.com/dotnet/ClangSharp), exists to help with generating callable native signatures and projecting data types into C#. The [Paint.NET](https://www.getpaint.net/) product successfully transitioned away from C++/CLI using ClangSharp.
+
+### Example
+
+Consider the following C++ class declaration and how to consume it via P/Invokes in C#.
+
+```cpp
+class myclass
+{
+public:
+    myclass() = default;
+    int get_value()
+    {
+        return -1;
+    }
+};
+
+// New C exports
+extern "C" __declspec(dllexport) myclass* new_myclass()
+{
+    return new myclass;
+}
+
+extern "C" __declspec(dllexport) void delete_myclass(myclass* mc)
+{
+    delete mc;
+}
+
+extern "C" __declspec(dllexport) int call_get_value(myclass* mc)
+{
+    return mc->get_value();
+}
+```
+
+Consumption of the `myclass` type can then occur in C# using P/Invokes.
+
+```csharp
+void* mc = new_myclass();
+Debug.Assert(mc != null);
+Console.WriteLine(call_get_value(mc));
+delete_myclass(mc);
+
+[LibraryImport("...")]
+static unsafe partial void* new_myclass();
+[LibraryImport("...")]
+static unsafe partial void delete_myclass(void* mc);
+[LibraryImport("...")]
+static unsafe partial int call_get_value(void* mc);
+```
+
+### Concerns
+
+The wrapping of complex or large C++ APIs to C is tedious and error prone. Source generation techniques can be employed to help create wrapped C APIs, but even this may represent significant effort.
+
+## Expose C++ as `IUnknown` based APIs
+
+Instead of reducing object-oriented concepts down to what can be expressed in C, using [COM interop](/dotnet/standard/native-interop/cominterop) permits keeping some aspects of C++. The .NET platform has a long history of reducing friction when using COM style interfaces. There is existing tools that can help generate COM interface definitions and [cross-platform support](/dotnet/standard/native-interop/tutorial-comwrappers) through the [`ComWrappers`](/dotnet/api/system.runtime.interopservices.comwrappers) API for `IUnknown` interop was enabled in .NET 6.
```

Comment:
```suggestion
Instead of reducing object-oriented concepts down to what can be expressed in C, using [COM interop](/dotnet/standard/native-interop/cominterop) permits keeping some aspects of C++. The .NET platform has a long history of reducing friction when using COM style interfaces. There are tools that can help generate COM interface definitions and [cross-platform support](/dotnet/standard/native-interop/tutorial-comwrappers) through the [`ComWrappers`](/dotnet/api/system.runtime.interopservices.comwrappers) API for `IUnknown` interop.
```

Not necessary to mention .NET 6. It is now the oldest version we support.